flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/calls/ScalarOperatorGens.scala

```diff
@@ -1369,11 +1369,12 @@ object ScalarOperatorGens {
       }
     }
 
-    val elementsCode = elements
+    val elementsCode = elements.zipWithIndex
       .map {
-        element =>
+        case (element, index) =>
```

Comment:
Is index unused here?  Why was the zipWithIndex modification added?

Reply:
it's used a couple of rows below
```scala
${if (index == 0) s"$tmpResult = ${castIfNumeric(elements.head)};" else ""}
```